{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T19:46:52.609080Z",
     "iopub.status.busy": "2023-11-08T19:46:52.608935Z",
     "iopub.status.idle": "2023-11-08T19:46:52.697999Z",
     "shell.execute_reply": "2023-11-08T19:46:52.697396Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T19:46:52.700556Z",
     "iopub.status.busy": "2023-11-08T19:46:52.700390Z",
     "iopub.status.idle": "2023-11-08T21:37:48.704631Z",
     "shell.execute_reply": "2023-11-08T21:37:48.704078Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:46:52.876801: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35069\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:16.048733: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:16.060248: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:16.060490: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:16.064568: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:16.064763: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:16.064931: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.185389: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.185998: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.186691: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.186844: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 14:48:46.187224: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:47.104060: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.104298: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.104471: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.104679: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.104848: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.104986: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:41:00.0, compute capability: 8.6\n",
      "2023-11-08 14:48:47.105014: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 14:48:47.126082: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 14:48:47.340934: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/bias/Assign' id:3510 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/bias, lstm_21/lstm_cell_21/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:48:47.487149: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:48:47.520389: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31555, 95)\n",
      "Train on 31555 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:51.840730: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:55.594607: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 14:48:57.713893: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n",
      "2023-11-08 14:48:57.927899: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 3.1518"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 14:49:20.766900: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.88768, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 30s 964us/sample - loss: 3.1518 - val_loss: 1.8877\n",
      "Epoch 2/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.7371\n",
      "Epoch 2: val_loss improved from 1.88768 to 1.54895, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 670us/sample - loss: 1.7371 - val_loss: 1.5489\n",
      "Epoch 3/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.5768\n",
      "Epoch 3: val_loss improved from 1.54895 to 1.48493, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 23s 718us/sample - loss: 1.5768 - val_loss: 1.4849\n",
      "Epoch 4/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.5282\n",
      "Epoch 4: val_loss improved from 1.48493 to 1.46285, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 23s 722us/sample - loss: 1.5282 - val_loss: 1.4629\n",
      "Epoch 5/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.5004\n",
      "Epoch 5: val_loss improved from 1.46285 to 1.44114, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 23s 720us/sample - loss: 1.5004 - val_loss: 1.4411\n",
      "Epoch 6/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4867\n",
      "Epoch 6: val_loss improved from 1.44114 to 1.42795, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 23s 720us/sample - loss: 1.4867 - val_loss: 1.4279\n",
      "Epoch 7/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4723\n",
      "Epoch 7: val_loss improved from 1.42795 to 1.41882, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 674us/sample - loss: 1.4723 - val_loss: 1.4188\n",
      "Epoch 8/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4628\n",
      "Epoch 8: val_loss improved from 1.41882 to 1.40761, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 650us/sample - loss: 1.4628 - val_loss: 1.4076\n",
      "Epoch 9/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4615\n",
      "Epoch 9: val_loss improved from 1.40761 to 1.40521, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 631us/sample - loss: 1.4615 - val_loss: 1.4052\n",
      "Epoch 10/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4566\n",
      "Epoch 10: val_loss improved from 1.40521 to 1.39798, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 627us/sample - loss: 1.4566 - val_loss: 1.3980\n",
      "Epoch 11/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4523\n",
      "Epoch 11: val_loss improved from 1.39798 to 1.39599, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 626us/sample - loss: 1.4523 - val_loss: 1.3960\n",
      "Epoch 12/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4654\n",
      "Epoch 12: val_loss did not improve from 1.39599\n",
      "31555/31555 [==============================] - 20s 618us/sample - loss: 1.4654 - val_loss: 1.3996\n",
      "Epoch 13/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4704\n",
      "Epoch 13: val_loss improved from 1.39599 to 1.39546, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 683us/sample - loss: 1.4704 - val_loss: 1.3955\n",
      "Epoch 14/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4509\n",
      "Epoch 14: val_loss did not improve from 1.39546\n",
      "31555/31555 [==============================] - 23s 715us/sample - loss: 1.4509 - val_loss: 1.3956\n",
      "Epoch 15/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.5198\n",
      "Epoch 15: val_loss did not improve from 1.39546\n",
      "31555/31555 [==============================] - 22s 707us/sample - loss: 1.5198 - val_loss: 1.4040\n",
      "Epoch 16/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4620\n",
      "Epoch 16: val_loss improved from 1.39546 to 1.39336, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 640us/sample - loss: 1.4620 - val_loss: 1.3934\n",
      "Epoch 17/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4751\n",
      "Epoch 17: val_loss did not improve from 1.39336\n",
      "31555/31555 [==============================] - 22s 693us/sample - loss: 1.4751 - val_loss: 1.4041\n",
      "Epoch 18/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4545\n",
      "Epoch 18: val_loss improved from 1.39336 to 1.39134, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 676us/sample - loss: 1.4545 - val_loss: 1.3913\n",
      "Epoch 19/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4506\n",
      "Epoch 19: val_loss did not improve from 1.39134\n",
      "31555/31555 [==============================] - 20s 638us/sample - loss: 1.4506 - val_loss: 1.3933\n",
      "Epoch 20/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4542\n",
      "Epoch 20: val_loss did not improve from 1.39134\n",
      "31555/31555 [==============================] - 22s 711us/sample - loss: 1.4542 - val_loss: 1.3995\n",
      "Epoch 21/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4474\n",
      "Epoch 21: val_loss improved from 1.39134 to 1.38733, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 672us/sample - loss: 1.4474 - val_loss: 1.3873\n",
      "Epoch 22/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4580\n",
      "Epoch 22: val_loss did not improve from 1.38733\n",
      "31555/31555 [==============================] - 20s 629us/sample - loss: 1.4580 - val_loss: 1.3995\n",
      "Epoch 23/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4467\n",
      "Epoch 23: val_loss did not improve from 1.38733\n",
      "31555/31555 [==============================] - 20s 618us/sample - loss: 1.4467 - val_loss: 1.3947\n",
      "Epoch 24/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4310\n",
      "Epoch 24: val_loss did not improve from 1.38733\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 1.4310 - val_loss: 1.3908\n",
      "Epoch 25/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4263\n",
      "Epoch 25: val_loss improved from 1.38733 to 1.38392, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 628us/sample - loss: 1.4263 - val_loss: 1.3839\n",
      "Epoch 26/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4248\n",
      "Epoch 26: val_loss improved from 1.38392 to 1.37867, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 641us/sample - loss: 1.4248 - val_loss: 1.3787\n",
      "Epoch 27/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4193\n",
      "Epoch 27: val_loss did not improve from 1.37867\n",
      "31555/31555 [==============================] - 19s 614us/sample - loss: 1.4193 - val_loss: 1.3787\n",
      "Epoch 28/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4159\n",
      "Epoch 28: val_loss improved from 1.37867 to 1.37865, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 19s 618us/sample - loss: 1.4159 - val_loss: 1.3786\n",
      "Epoch 29/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4137\n",
      "Epoch 29: val_loss improved from 1.37865 to 1.37650, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 1.4137 - val_loss: 1.3765\n",
      "Epoch 30/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4121\n",
      "Epoch 30: val_loss improved from 1.37650 to 1.37303, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 1.4121 - val_loss: 1.3730\n",
      "Epoch 31/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4079\n",
      "Epoch 31: val_loss improved from 1.37303 to 1.37039, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 638us/sample - loss: 1.4079 - val_loss: 1.3704\n",
      "Epoch 32/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4053\n",
      "Epoch 32: val_loss improved from 1.37039 to 1.36725, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 19s 615us/sample - loss: 1.4053 - val_loss: 1.3673\n",
      "Epoch 33/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4029\n",
      "Epoch 33: val_loss did not improve from 1.36725\n",
      "31555/31555 [==============================] - 19s 610us/sample - loss: 1.4029 - val_loss: 1.3678\n",
      "Epoch 34/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4036\n",
      "Epoch 34: val_loss improved from 1.36725 to 1.36178, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 19s 616us/sample - loss: 1.4036 - val_loss: 1.3618\n",
      "Epoch 35/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3991\n",
      "Epoch 35: val_loss did not improve from 1.36178\n",
      "31555/31555 [==============================] - 22s 695us/sample - loss: 1.3991 - val_loss: 1.3647\n",
      "Epoch 36/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3954\n",
      "Epoch 36: val_loss did not improve from 1.36178\n",
      "31555/31555 [==============================] - 23s 715us/sample - loss: 1.3954 - val_loss: 1.3631\n",
      "Epoch 37/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3955\n",
      "Epoch 37: val_loss improved from 1.36178 to 1.36054, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 23s 720us/sample - loss: 1.3955 - val_loss: 1.3605\n",
      "Epoch 38/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3926\n",
      "Epoch 38: val_loss improved from 1.36054 to 1.35628, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 23s 715us/sample - loss: 1.3926 - val_loss: 1.3563\n",
      "Epoch 39/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3900\n",
      "Epoch 39: val_loss did not improve from 1.35628\n",
      "31555/31555 [==============================] - 21s 651us/sample - loss: 1.3900 - val_loss: 1.3567\n",
      "Epoch 40/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3869\n",
      "Epoch 40: val_loss did not improve from 1.35628\n",
      "31555/31555 [==============================] - 19s 610us/sample - loss: 1.3869 - val_loss: 1.3566\n",
      "Epoch 41/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3873\n",
      "Epoch 41: val_loss did not improve from 1.35628\n",
      "31555/31555 [==============================] - 19s 615us/sample - loss: 1.3873 - val_loss: 1.3583\n",
      "Epoch 42/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3877\n",
      "Epoch 42: val_loss did not improve from 1.35628\n",
      "31555/31555 [==============================] - 19s 615us/sample - loss: 1.3877 - val_loss: 1.3567\n",
      "Epoch 43/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3877\n",
      "Epoch 43: val_loss improved from 1.35628 to 1.35204, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 1.3877 - val_loss: 1.3520\n",
      "Epoch 44/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3824\n",
      "Epoch 44: val_loss did not improve from 1.35204\n",
      "31555/31555 [==============================] - 20s 625us/sample - loss: 1.3824 - val_loss: 1.3562\n",
      "Epoch 45/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3816\n",
      "Epoch 45: val_loss did not improve from 1.35204\n",
      "31555/31555 [==============================] - 19s 614us/sample - loss: 1.3816 - val_loss: 1.3552\n",
      "Epoch 46/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3792\n",
      "Epoch 46: val_loss did not improve from 1.35204\n",
      "31555/31555 [==============================] - 19s 611us/sample - loss: 1.3792 - val_loss: 1.3536\n",
      "Epoch 47/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3792\n",
      "Epoch 47: val_loss improved from 1.35204 to 1.34884, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 627us/sample - loss: 1.3792 - val_loss: 1.3488\n",
      "Epoch 48/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3783\n",
      "Epoch 48: val_loss did not improve from 1.34884\n",
      "31555/31555 [==============================] - 19s 610us/sample - loss: 1.3783 - val_loss: 1.3513\n",
      "Epoch 49/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3763\n",
      "Epoch 49: val_loss did not improve from 1.34884\n",
      "31555/31555 [==============================] - 20s 639us/sample - loss: 1.3763 - val_loss: 1.3528\n",
      "Epoch 50/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3756\n",
      "Epoch 50: val_loss improved from 1.34884 to 1.34820, saving model to ./checkpoints/unknown_person_baseline_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 710us/sample - loss: 1.3756 - val_loss: 1.3482\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:06:16.658684: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_29_1/lstm_cell_66/recurrent_kernel/Assign' id:23302 op device:{requested: '', assigned: ''} def:{{{node lstm_29_1/lstm_cell_66/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_29_1/lstm_cell_66/recurrent_kernel, lstm_29_1/lstm_cell_66/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:06:18.296091: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_12_1/lstm_cell_49/kernel/v/Assign' id:26075 op device:{requested: '', assigned: ''} def:{{{node lstm_12_1/lstm_cell_49/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_12_1/lstm_cell_49/kernel/v, lstm_12_1/lstm_cell_49/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 15:06:19.598779: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1932)\n",
      "(1514, 1932)\n",
      "(1644, 1932)\n",
      "(1764, 1932)\n",
      "(1836, 1932)\n",
      "(1699, 1932)\n",
      "(1369, 1932)\n",
      "(1766, 1932)\n",
      "(1631, 1932)\n",
      "(1692, 1932)\n",
      "(1550, 1932)\n",
      "(1920, 1932)\n",
      "(1739, 1932)\n",
      "(1788, 1932)\n",
      "(1812, 1932)\n",
      "(1824, 1932)\n",
      "(970, 1932)\n",
      "(1656, 1932)\n",
      "(1896, 1932)\n",
      "{1: 6.420477290145464, 2: 5.459450545952956, 4: 10.0, 5: 5.649065996828305, 6: 6.163528129725273, 8: 9.068458807618333, 9: 6.751926029474841, 10: 8.033685446424467, 11: 7.39080693327163, 12: 8.960112317828333, 13: 7.100463730364462, 17: 9.518077097762282, 19: 9.283534919719933, 21: 9.065605191112583, 22: 1.0, 26: 6.106064768667812, 27: 5.074488767459354, 28: 6.30438475019741, 29: 2.410489531333754}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2792505/616055324.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31555 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:10:23.348809: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 9.8077\n",
      "Epoch 1: val_loss improved from inf to 1.36455, saving model to ./checkpoints/unknown_person_p25_20.h5\n",
      "31555/31555 [==============================] - 26s 813us/sample - loss: 9.8077 - val_loss: 1.3645\n",
      "Epoch 2/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.7683\n",
      "Epoch 2: val_loss improved from 1.36455 to 1.35240, saving model to ./checkpoints/unknown_person_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 708us/sample - loss: 9.7683 - val_loss: 1.3524\n",
      "Epoch 3/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.7398\n",
      "Epoch 3: val_loss did not improve from 1.35240\n",
      "31555/31555 [==============================] - 22s 706us/sample - loss: 9.7398 - val_loss: 1.3589\n",
      "Epoch 4/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.6765\n",
      "Epoch 4: val_loss improved from 1.35240 to 1.35057, saving model to ./checkpoints/unknown_person_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 708us/sample - loss: 9.6765 - val_loss: 1.3506\n",
      "Epoch 5/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.6495\n",
      "Epoch 5: val_loss did not improve from 1.35057\n",
      "31555/31555 [==============================] - 22s 710us/sample - loss: 9.6495 - val_loss: 1.3579\n",
      "Epoch 6/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.6406\n",
      "Epoch 6: val_loss improved from 1.35057 to 1.34542, saving model to ./checkpoints/unknown_person_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 713us/sample - loss: 9.6406 - val_loss: 1.3454\n",
      "Epoch 7/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.6211\n",
      "Epoch 7: val_loss did not improve from 1.34542\n",
      "31555/31555 [==============================] - 22s 705us/sample - loss: 9.6211 - val_loss: 1.3627\n",
      "Epoch 8/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.6047\n",
      "Epoch 8: val_loss did not improve from 1.34542\n",
      "31555/31555 [==============================] - 21s 674us/sample - loss: 9.6047 - val_loss: 1.3545\n",
      "Epoch 9/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.5869\n",
      "Epoch 9: val_loss did not improve from 1.34542\n",
      "31555/31555 [==============================] - 19s 610us/sample - loss: 9.5869 - val_loss: 1.3477\n",
      "Epoch 10/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.5632\n",
      "Epoch 10: val_loss did not improve from 1.34542\n",
      "31555/31555 [==============================] - 21s 661us/sample - loss: 9.5632 - val_loss: 1.3521\n",
      "Epoch 11/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.5505\n",
      "Epoch 11: val_loss improved from 1.34542 to 1.34188, saving model to ./checkpoints/unknown_person_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 682us/sample - loss: 9.5505 - val_loss: 1.3419\n",
      "Epoch 12/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.5368\n",
      "Epoch 12: val_loss did not improve from 1.34188\n",
      "31555/31555 [==============================] - 22s 696us/sample - loss: 9.5368 - val_loss: 1.3476\n",
      "Epoch 13/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.5329\n",
      "Epoch 13: val_loss improved from 1.34188 to 1.33643, saving model to ./checkpoints/unknown_person_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 668us/sample - loss: 9.5329 - val_loss: 1.3364\n",
      "Epoch 14/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.5249\n",
      "Epoch 14: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 22s 684us/sample - loss: 9.5249 - val_loss: 1.3432\n",
      "Epoch 15/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.4945\n",
      "Epoch 15: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 9.4945 - val_loss: 1.3369\n",
      "Epoch 16/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.4870\n",
      "Epoch 16: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 22s 706us/sample - loss: 9.4870 - val_loss: 1.3393\n",
      "Epoch 17/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.4975\n",
      "Epoch 17: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 20s 649us/sample - loss: 9.4975 - val_loss: 1.3494\n",
      "Epoch 18/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.4675\n",
      "Epoch 18: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 22s 697us/sample - loss: 9.4675 - val_loss: 1.3397\n",
      "Epoch 19/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.4583\n",
      "Epoch 19: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 23s 714us/sample - loss: 9.4583 - val_loss: 1.3416\n",
      "Epoch 20/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.4734\n",
      "Epoch 20: val_loss did not improve from 1.33643\n",
      "31555/31555 [==============================] - 23s 716us/sample - loss: 9.4734 - val_loss: 1.3371\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:17:44.525399: W tensorflow/c/c_api.cc:304] Operation '{name:'conv2d_2_2/bias/Assign' id:37819 op device:{requested: '', assigned: ''} def:{{{node conv2d_2_2/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](conv2d_2_2/bias, conv2d_2_2/bias/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:17:46.983278: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_14_2/lstm_cell_88/recurrent_kernel/v/Assign' id:45508 op device:{requested: '', assigned: ''} def:{{{node lstm_14_2/lstm_cell_88/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_14_2/lstm_cell_88/recurrent_kernel/v, lstm_14_2/lstm_cell_88/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31555 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:17:52.122961: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:18:04.968672: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3720"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:18:26.619071: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.34908, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 28s 886us/sample - loss: 1.3720 - val_loss: 1.3491\n",
      "Epoch 2/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3701\n",
      "Epoch 2: val_loss improved from 1.34908 to 1.34819, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 21s 650us/sample - loss: 1.3701 - val_loss: 1.3482\n",
      "Epoch 3/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3715\n",
      "Epoch 3: val_loss did not improve from 1.34819\n",
      "31555/31555 [==============================] - 23s 717us/sample - loss: 1.3715 - val_loss: 1.3498\n",
      "Epoch 4/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3654\n",
      "Epoch 4: val_loss did not improve from 1.34819\n",
      "31555/31555 [==============================] - 19s 613us/sample - loss: 1.3654 - val_loss: 1.3510\n",
      "Epoch 5/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3658\n",
      "Epoch 5: val_loss improved from 1.34819 to 1.34563, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 19s 616us/sample - loss: 1.3658 - val_loss: 1.3456\n",
      "Epoch 6/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3631\n",
      "Epoch 6: val_loss did not improve from 1.34563\n",
      "31555/31555 [==============================] - 19s 613us/sample - loss: 1.3631 - val_loss: 1.3473\n",
      "Epoch 7/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3643\n",
      "Epoch 7: val_loss improved from 1.34563 to 1.34554, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 618us/sample - loss: 1.3643 - val_loss: 1.3455\n",
      "Epoch 8/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3614\n",
      "Epoch 8: val_loss improved from 1.34554 to 1.34073, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 1.3614 - val_loss: 1.3407\n",
      "Epoch 9/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3634\n",
      "Epoch 9: val_loss did not improve from 1.34073\n",
      "31555/31555 [==============================] - 22s 701us/sample - loss: 1.3634 - val_loss: 1.3463\n",
      "Epoch 10/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3651\n",
      "Epoch 10: val_loss did not improve from 1.34073\n",
      "31555/31555 [==============================] - 21s 660us/sample - loss: 1.3651 - val_loss: 1.3488\n",
      "Epoch 11/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3587\n",
      "Epoch 11: val_loss did not improve from 1.34073\n",
      "31555/31555 [==============================] - 19s 611us/sample - loss: 1.3587 - val_loss: 1.3413\n",
      "Epoch 12/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3581\n",
      "Epoch 12: val_loss improved from 1.34073 to 1.33662, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 19s 616us/sample - loss: 1.3581 - val_loss: 1.3366\n",
      "Epoch 13/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3575\n",
      "Epoch 13: val_loss did not improve from 1.33662\n",
      "31555/31555 [==============================] - 22s 689us/sample - loss: 1.3575 - val_loss: 1.3420\n",
      "Epoch 14/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3537\n",
      "Epoch 14: val_loss improved from 1.33662 to 1.33442, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 713us/sample - loss: 1.3537 - val_loss: 1.3344\n",
      "Epoch 15/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3532\n",
      "Epoch 15: val_loss did not improve from 1.33442\n",
      "31555/31555 [==============================] - 20s 623us/sample - loss: 1.3532 - val_loss: 1.3419\n",
      "Epoch 16/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3541\n",
      "Epoch 16: val_loss improved from 1.33442 to 1.33316, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 19s 616us/sample - loss: 1.3541 - val_loss: 1.3332\n",
      "Epoch 17/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3513\n",
      "Epoch 17: val_loss did not improve from 1.33316\n",
      "31555/31555 [==============================] - 19s 604us/sample - loss: 1.3513 - val_loss: 1.3366\n",
      "Epoch 18/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 18: val_loss did not improve from 1.33316\n",
      "31555/31555 [==============================] - 20s 642us/sample - loss: 1.3489 - val_loss: 1.3369\n",
      "Epoch 19/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3485\n",
      "Epoch 19: val_loss improved from 1.33316 to 1.33126, saving model to ./checkpoints/unknown_person_baseline_2_p25_20.h5\n",
      "31555/31555 [==============================] - 22s 699us/sample - loss: 1.3485 - val_loss: 1.3313\n",
      "Epoch 20/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3490\n",
      "Epoch 20: val_loss did not improve from 1.33126\n",
      "31555/31555 [==============================] - 22s 698us/sample - loss: 1.3490 - val_loss: 1.3317\n",
      "Train on 31555 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3471\n",
      "Epoch 1: val_loss improved from inf to 1.33170, saving model to ./checkpoints/unknown_person_baseline_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 704us/sample - loss: 1.3471 - val_loss: 1.3317\n",
      "Epoch 2/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3489\n",
      "Epoch 2: val_loss improved from 1.33170 to 1.33045, saving model to ./checkpoints/unknown_person_baseline_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 708us/sample - loss: 1.3489 - val_loss: 1.3305\n",
      "Epoch 3/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3470\n",
      "Epoch 3: val_loss did not improve from 1.33045\n",
      "31555/31555 [==============================] - 20s 639us/sample - loss: 1.3470 - val_loss: 1.3313\n",
      "Epoch 4/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3416\n",
      "Epoch 4: val_loss did not improve from 1.33045\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 1.3416 - val_loss: 1.3355\n",
      "Epoch 5/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3399\n",
      "Epoch 5: val_loss improved from 1.33045 to 1.32827, saving model to ./checkpoints/unknown_person_baseline_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 697us/sample - loss: 1.3399 - val_loss: 1.3283\n",
      "Epoch 6/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3394\n",
      "Epoch 6: val_loss improved from 1.32827 to 1.32818, saving model to ./checkpoints/unknown_person_baseline_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 1.3394 - val_loss: 1.3282\n",
      "Epoch 7/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3421\n",
      "Epoch 7: val_loss improved from 1.32818 to 1.32770, saving model to ./checkpoints/unknown_person_baseline_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 709us/sample - loss: 1.3421 - val_loss: 1.3277\n",
      "Epoch 8/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3481\n",
      "Epoch 8: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 627us/sample - loss: 1.3481 - val_loss: 1.3406\n",
      "Epoch 9/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3464\n",
      "Epoch 9: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 616us/sample - loss: 1.3464 - val_loss: 1.3370\n",
      "Epoch 10/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3581\n",
      "Epoch 10: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 616us/sample - loss: 1.3581 - val_loss: 1.3311\n",
      "Epoch 11/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3593\n",
      "Epoch 11: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 633us/sample - loss: 1.3593 - val_loss: 1.3371\n",
      "Epoch 12/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3518\n",
      "Epoch 12: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 608us/sample - loss: 1.3518 - val_loss: 1.3335\n",
      "Epoch 13/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3516\n",
      "Epoch 13: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 648us/sample - loss: 1.3516 - val_loss: 1.3302\n",
      "Epoch 14/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3905\n",
      "Epoch 14: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 705us/sample - loss: 1.3905 - val_loss: 1.3396\n",
      "Epoch 15/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3609\n",
      "Epoch 15: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 611us/sample - loss: 1.3609 - val_loss: 1.3343\n",
      "Epoch 16/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3822\n",
      "Epoch 16: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 621us/sample - loss: 1.3822 - val_loss: 1.3363\n",
      "Epoch 17/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3854\n",
      "Epoch 17: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 602us/sample - loss: 1.3854 - val_loss: 1.3398\n",
      "Epoch 18/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4077\n",
      "Epoch 18: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 597us/sample - loss: 1.4077 - val_loss: 1.3413\n",
      "Epoch 19/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4081\n",
      "Epoch 19: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 598us/sample - loss: 1.4081 - val_loss: 1.3424\n",
      "Epoch 20/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4192\n",
      "Epoch 20: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 589us/sample - loss: 1.4192 - val_loss: 1.3494\n",
      "Epoch 21/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4154\n",
      "Epoch 21: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 712us/sample - loss: 1.4154 - val_loss: 1.3557\n",
      "Epoch 22/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.5236\n",
      "Epoch 22: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 706us/sample - loss: 1.5236 - val_loss: 1.3723\n",
      "Epoch 23/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4390\n",
      "Epoch 23: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 692us/sample - loss: 1.4390 - val_loss: 1.3652\n",
      "Epoch 24/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4181\n",
      "Epoch 24: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 621us/sample - loss: 1.4181 - val_loss: 1.3543\n",
      "Epoch 25/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3969\n",
      "Epoch 25: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 636us/sample - loss: 1.3969 - val_loss: 1.3589\n",
      "Epoch 26/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3901\n",
      "Epoch 26: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 640us/sample - loss: 1.3901 - val_loss: 1.3548\n",
      "Epoch 27/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3844\n",
      "Epoch 27: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 700us/sample - loss: 1.3844 - val_loss: 1.3524\n",
      "Epoch 28/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3851\n",
      "Epoch 28: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 699us/sample - loss: 1.3851 - val_loss: 1.3515\n",
      "Epoch 29/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3763\n",
      "Epoch 29: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 701us/sample - loss: 1.3763 - val_loss: 1.3551\n",
      "Epoch 30/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3748\n",
      "Epoch 30: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 695us/sample - loss: 1.3748 - val_loss: 1.3488\n",
      "Epoch 31/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3713\n",
      "Epoch 31: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 683us/sample - loss: 1.3713 - val_loss: 1.3459\n",
      "Epoch 32/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3699\n",
      "Epoch 32: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 702us/sample - loss: 1.3699 - val_loss: 1.3505\n",
      "Epoch 33/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3683\n",
      "Epoch 33: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 1.3683 - val_loss: 1.3452\n",
      "Epoch 34/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3649\n",
      "Epoch 34: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 701us/sample - loss: 1.3649 - val_loss: 1.3422\n",
      "Epoch 35/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 35: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 21s 662us/sample - loss: 1.3650 - val_loss: 1.3460\n",
      "Epoch 36/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3625\n",
      "Epoch 36: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 608us/sample - loss: 1.3625 - val_loss: 1.3460\n",
      "Epoch 37/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3608\n",
      "Epoch 37: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 21s 675us/sample - loss: 1.3608 - val_loss: 1.3440\n",
      "Epoch 38/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3589\n",
      "Epoch 38: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 707us/sample - loss: 1.3589 - val_loss: 1.3459\n",
      "Epoch 39/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3583\n",
      "Epoch 39: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 690us/sample - loss: 1.3583 - val_loss: 1.3468\n",
      "Epoch 40/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3575\n",
      "Epoch 40: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 702us/sample - loss: 1.3575 - val_loss: 1.3426\n",
      "Epoch 41/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3564\n",
      "Epoch 41: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 21s 679us/sample - loss: 1.3564 - val_loss: 1.3389\n",
      "Epoch 42/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3553\n",
      "Epoch 42: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 617us/sample - loss: 1.3553 - val_loss: 1.3420\n",
      "Epoch 43/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3550\n",
      "Epoch 43: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 637us/sample - loss: 1.3550 - val_loss: 1.3421\n",
      "Epoch 44/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3568\n",
      "Epoch 44: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 20s 640us/sample - loss: 1.3568 - val_loss: 1.3390\n",
      "Epoch 45/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3533\n",
      "Epoch 45: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 19s 602us/sample - loss: 1.3533 - val_loss: 1.3397\n",
      "Epoch 46/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3498\n",
      "Epoch 46: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 698us/sample - loss: 1.3498 - val_loss: 1.3368\n",
      "Epoch 47/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3516\n",
      "Epoch 47: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 704us/sample - loss: 1.3516 - val_loss: 1.3416\n",
      "Epoch 48/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3507\n",
      "Epoch 48: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 1.3507 - val_loss: 1.3370\n",
      "Epoch 49/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3496\n",
      "Epoch 49: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 21s 663us/sample - loss: 1.3496 - val_loss: 1.3356\n",
      "Epoch 50/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3458\n",
      "Epoch 50: val_loss did not improve from 1.32770\n",
      "31555/31555 [==============================] - 21s 679us/sample - loss: 1.3458 - val_loss: 1.3344\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:42:30.915042: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_2_3/kernel/Assign' id:63380 op device:{requested: '', assigned: ''} def:{{{node dense_2_3/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_2_3/kernel, dense_2_3/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:42:34.141973: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_19_3/lstm_cell_130/bias/m/Assign' id:64305 op device:{requested: '', assigned: ''} def:{{{node lstm_19_3/lstm_cell_130/bias/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_19_3/lstm_cell_130/bias/m, lstm_19_3/lstm_cell_130/bias/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:42:37.220590: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1932)\n",
      "(1514, 1932)\n",
      "(1644, 1932)\n",
      "(1764, 1932)\n",
      "(1836, 1932)\n",
      "(1699, 1932)\n",
      "(1369, 1932)\n",
      "(1766, 1932)\n",
      "(1631, 1932)\n",
      "(1692, 1932)\n",
      "(1550, 1932)\n",
      "(1920, 1932)\n",
      "(1739, 1932)\n",
      "(1788, 1932)\n",
      "(1812, 1932)\n",
      "(1824, 1932)\n",
      "(970, 1932)\n",
      "(1656, 1932)\n",
      "(1896, 1932)\n",
      "{1: 5.9210177475080314, 2: 4.439995537428065, 4: 10.0, 5: 5.543503487921628, 6: 5.923247276557354, 8: 8.802503959654132, 9: 6.307406200339612, 10: 7.959947538425995, 11: 7.001701575164908, 12: 8.755014463828132, 13: 6.552178270964205, 17: 9.323778041510703, 19: 9.132518181906027, 21: 8.795339247705245, 22: 1.0, 26: 5.324725020358288, 27: 4.584786670613876, 28: 5.7543090612171754, 29: 1.2246896184588842}\n",
      "Train on 31555 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:46:21.475859: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 9.1551\n",
      "Epoch 1: val_loss improved from inf to 1.34043, saving model to ./checkpoints/unknown_person_p25_21.h5\n",
      "31555/31555 [==============================] - 28s 884us/sample - loss: 9.1551 - val_loss: 1.3404\n",
      "Epoch 2/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.1018\n",
      "Epoch 2: val_loss did not improve from 1.34043\n",
      "31555/31555 [==============================] - 19s 590us/sample - loss: 9.1018 - val_loss: 1.3413\n",
      "Epoch 3/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0787\n",
      "Epoch 3: val_loss did not improve from 1.34043\n",
      "31555/31555 [==============================] - 19s 593us/sample - loss: 9.0787 - val_loss: 1.3454\n",
      "Epoch 4/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0726\n",
      "Epoch 4: val_loss improved from 1.34043 to 1.33340, saving model to ./checkpoints/unknown_person_p25_21.h5\n",
      "31555/31555 [==============================] - 19s 595us/sample - loss: 9.0726 - val_loss: 1.3334\n",
      "Epoch 5/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0195\n",
      "Epoch 5: val_loss did not improve from 1.33340\n",
      "31555/31555 [==============================] - 19s 607us/sample - loss: 9.0195 - val_loss: 1.3398\n",
      "Epoch 6/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9775\n",
      "Epoch 6: val_loss did not improve from 1.33340\n",
      "31555/31555 [==============================] - 21s 668us/sample - loss: 8.9775 - val_loss: 1.3386\n",
      "Epoch 7/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9606\n",
      "Epoch 7: val_loss improved from 1.33340 to 1.33264, saving model to ./checkpoints/unknown_person_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 710us/sample - loss: 8.9606 - val_loss: 1.3326\n",
      "Epoch 8/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9670\n",
      "Epoch 8: val_loss did not improve from 1.33264\n",
      "31555/31555 [==============================] - 22s 706us/sample - loss: 8.9670 - val_loss: 1.3364\n",
      "Epoch 9/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9357\n",
      "Epoch 9: val_loss improved from 1.33264 to 1.32814, saving model to ./checkpoints/unknown_person_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 711us/sample - loss: 8.9357 - val_loss: 1.3281\n",
      "Epoch 10/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9230\n",
      "Epoch 10: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 22s 707us/sample - loss: 8.9230 - val_loss: 1.3377\n",
      "Epoch 11/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9251\n",
      "Epoch 11: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 20s 638us/sample - loss: 8.9251 - val_loss: 1.3368\n",
      "Epoch 12/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9116\n",
      "Epoch 12: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 22s 684us/sample - loss: 8.9116 - val_loss: 1.3319\n",
      "Epoch 13/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9106\n",
      "Epoch 13: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 21s 681us/sample - loss: 8.9106 - val_loss: 1.3351\n",
      "Epoch 14/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8716\n",
      "Epoch 14: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 20s 639us/sample - loss: 8.8716 - val_loss: 1.3361\n",
      "Epoch 15/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8600\n",
      "Epoch 15: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 19s 608us/sample - loss: 8.8600 - val_loss: 1.3293\n",
      "Epoch 16/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8613\n",
      "Epoch 16: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 19s 606us/sample - loss: 8.8613 - val_loss: 1.3298\n",
      "Epoch 17/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8375\n",
      "Epoch 17: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 19s 598us/sample - loss: 8.8375 - val_loss: 1.3289\n",
      "Epoch 18/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8564\n",
      "Epoch 18: val_loss did not improve from 1.32814\n",
      "31555/31555 [==============================] - 19s 600us/sample - loss: 8.8564 - val_loss: 1.3342\n",
      "Epoch 19/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8382\n",
      "Epoch 19: val_loss improved from 1.32814 to 1.32668, saving model to ./checkpoints/unknown_person_p25_21.h5\n",
      "31555/31555 [==============================] - 20s 620us/sample - loss: 8.8382 - val_loss: 1.3267\n",
      "Epoch 20/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.8302\n",
      "Epoch 20: val_loss did not improve from 1.32668\n",
      "31555/31555 [==============================] - 20s 637us/sample - loss: 8.8302 - val_loss: 1.3279\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:53:16.728226: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_30_4/lstm_cell_178/kernel/Assign' id:81598 op device:{requested: '', assigned: ''} def:{{{node lstm_30_4/lstm_cell_178/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_30_4/lstm_cell_178/kernel, lstm_30_4/lstm_cell_178/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:53:20.961533: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_12_4/lstm_cell_160/recurrent_kernel/v/Assign' id:84236 op device:{requested: '', assigned: ''} def:{{{node lstm_12_4/lstm_cell_160/recurrent_kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_12_4/lstm_cell_160/recurrent_kernel/v, lstm_12_4/lstm_cell_160/recurrent_kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31555 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:53:27.693850: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:53:49.806205: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3489"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:54:10.654273: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.33002, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 30s 953us/sample - loss: 1.3489 - val_loss: 1.3300\n",
      "Epoch 2/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3390\n",
      "Epoch 2: val_loss improved from 1.33002 to 1.32787, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 19s 606us/sample - loss: 1.3390 - val_loss: 1.3279\n",
      "Epoch 3/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3435\n",
      "Epoch 3: val_loss did not improve from 1.32787\n",
      "31555/31555 [==============================] - 19s 602us/sample - loss: 1.3435 - val_loss: 1.3306\n",
      "Epoch 4/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3389\n",
      "Epoch 4: val_loss improved from 1.32787 to 1.32493, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 20s 637us/sample - loss: 1.3389 - val_loss: 1.3249\n",
      "Epoch 5/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3382\n",
      "Epoch 5: val_loss did not improve from 1.32493\n",
      "31555/31555 [==============================] - 20s 639us/sample - loss: 1.3382 - val_loss: 1.3275\n",
      "Epoch 6/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3352\n",
      "Epoch 6: val_loss did not improve from 1.32493\n",
      "31555/31555 [==============================] - 21s 656us/sample - loss: 1.3352 - val_loss: 1.3328\n",
      "Epoch 7/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3360\n",
      "Epoch 7: val_loss did not improve from 1.32493\n",
      "31555/31555 [==============================] - 21s 650us/sample - loss: 1.3360 - val_loss: 1.3282\n",
      "Epoch 8/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3344\n",
      "Epoch 8: val_loss did not improve from 1.32493\n",
      "31555/31555 [==============================] - 22s 692us/sample - loss: 1.3344 - val_loss: 1.3303\n",
      "Epoch 9/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3323\n",
      "Epoch 9: val_loss improved from 1.32493 to 1.32385, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 21s 653us/sample - loss: 1.3323 - val_loss: 1.3239\n",
      "Epoch 10/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3330\n",
      "Epoch 10: val_loss did not improve from 1.32385\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 1.3330 - val_loss: 1.3252\n",
      "Epoch 11/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3319\n",
      "Epoch 11: val_loss improved from 1.32385 to 1.32339, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 20s 639us/sample - loss: 1.3319 - val_loss: 1.3234\n",
      "Epoch 12/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3307\n",
      "Epoch 12: val_loss did not improve from 1.32339\n",
      "31555/31555 [==============================] - 21s 673us/sample - loss: 1.3307 - val_loss: 1.3247\n",
      "Epoch 13/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3290\n",
      "Epoch 13: val_loss improved from 1.32339 to 1.31898, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 694us/sample - loss: 1.3290 - val_loss: 1.3190\n",
      "Epoch 14/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3295\n",
      "Epoch 14: val_loss did not improve from 1.31898\n",
      "31555/31555 [==============================] - 20s 631us/sample - loss: 1.3295 - val_loss: 1.3244\n",
      "Epoch 15/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3265\n",
      "Epoch 15: val_loss did not improve from 1.31898\n",
      "31555/31555 [==============================] - 19s 599us/sample - loss: 1.3265 - val_loss: 1.3240\n",
      "Epoch 16/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3261\n",
      "Epoch 16: val_loss did not improve from 1.31898\n",
      "31555/31555 [==============================] - 20s 622us/sample - loss: 1.3261 - val_loss: 1.3257\n",
      "Epoch 17/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3287\n",
      "Epoch 17: val_loss did not improve from 1.31898\n",
      "31555/31555 [==============================] - 20s 644us/sample - loss: 1.3287 - val_loss: 1.3243\n",
      "Epoch 18/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3261\n",
      "Epoch 18: val_loss improved from 1.31898 to 1.31846, saving model to ./checkpoints/unknown_person_baseline_2_p25_21.h5\n",
      "31555/31555 [==============================] - 22s 708us/sample - loss: 1.3261 - val_loss: 1.3185\n",
      "Epoch 19/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3253\n",
      "Epoch 19: val_loss did not improve from 1.31846\n",
      "31555/31555 [==============================] - 22s 698us/sample - loss: 1.3253 - val_loss: 1.3201\n",
      "Epoch 20/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3249\n",
      "Epoch 20: val_loss did not improve from 1.31846\n",
      "31555/31555 [==============================] - 22s 700us/sample - loss: 1.3249 - val_loss: 1.3213\n",
      "Train on 31555 samples, validate on 3514 samples\n",
      "Epoch 1/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3251\n",
      "Epoch 1: val_loss improved from inf to 1.32031, saving model to ./checkpoints/unknown_person_baseline_p25_22.h5\n",
      "31555/31555 [==============================] - 22s 688us/sample - loss: 1.3251 - val_loss: 1.3203\n",
      "Epoch 2/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3253\n",
      "Epoch 2: val_loss did not improve from 1.32031\n",
      "31555/31555 [==============================] - 20s 624us/sample - loss: 1.3253 - val_loss: 1.3223\n",
      "Epoch 3/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3218\n",
      "Epoch 3: val_loss improved from 1.32031 to 1.31908, saving model to ./checkpoints/unknown_person_baseline_p25_22.h5\n",
      "31555/31555 [==============================] - 19s 608us/sample - loss: 1.3218 - val_loss: 1.3191\n",
      "Epoch 4/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3232\n",
      "Epoch 4: val_loss improved from 1.31908 to 1.31827, saving model to ./checkpoints/unknown_person_baseline_p25_22.h5\n",
      "31555/31555 [==============================] - 20s 634us/sample - loss: 1.3232 - val_loss: 1.3183\n",
      "Epoch 5/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3175\n",
      "Epoch 5: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 698us/sample - loss: 1.3175 - val_loss: 1.3207\n",
      "Epoch 6/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3179\n",
      "Epoch 6: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 699us/sample - loss: 1.3179 - val_loss: 1.3277\n",
      "Epoch 7/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3182\n",
      "Epoch 7: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 684us/sample - loss: 1.3182 - val_loss: 1.3186\n",
      "Epoch 8/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3299\n",
      "Epoch 8: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 661us/sample - loss: 1.3299 - val_loss: 1.3220\n",
      "Epoch 9/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3316\n",
      "Epoch 9: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 1.3316 - val_loss: 1.3239\n",
      "Epoch 10/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3520\n",
      "Epoch 10: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 658us/sample - loss: 1.3520 - val_loss: 1.3261\n",
      "Epoch 11/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3373\n",
      "Epoch 11: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 593us/sample - loss: 1.3373 - val_loss: 1.3261\n",
      "Epoch 12/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3560\n",
      "Epoch 12: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 666us/sample - loss: 1.3560 - val_loss: 1.3296\n",
      "Epoch 13/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3830\n",
      "Epoch 13: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 683us/sample - loss: 1.3830 - val_loss: 1.3288\n",
      "Epoch 14/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3547\n",
      "Epoch 14: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 604us/sample - loss: 1.3547 - val_loss: 1.3291\n",
      "Epoch 15/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3801\n",
      "Epoch 15: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 596us/sample - loss: 1.3801 - val_loss: 1.3363\n",
      "Epoch 16/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4209\n",
      "Epoch 16: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 600us/sample - loss: 1.4209 - val_loss: 1.3379\n",
      "Epoch 17/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3790\n",
      "Epoch 17: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 601us/sample - loss: 1.3790 - val_loss: 1.3311\n",
      "Epoch 18/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3930\n",
      "Epoch 18: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 600us/sample - loss: 1.3930 - val_loss: 1.3405\n",
      "Epoch 19/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3827\n",
      "Epoch 19: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 660us/sample - loss: 1.3827 - val_loss: 1.3375\n",
      "Epoch 20/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.5278\n",
      "Epoch 20: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 599us/sample - loss: 1.5278 - val_loss: 1.3865\n",
      "Epoch 21/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4534\n",
      "Epoch 21: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 686us/sample - loss: 1.4534 - val_loss: 1.3681\n",
      "Epoch 22/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4377\n",
      "Epoch 22: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 704us/sample - loss: 1.4377 - val_loss: 1.3669\n",
      "Epoch 23/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4435\n",
      "Epoch 23: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 699us/sample - loss: 1.4435 - val_loss: 1.3714\n",
      "Epoch 24/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4190\n",
      "Epoch 24: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 635us/sample - loss: 1.4190 - val_loss: 1.3688\n",
      "Epoch 25/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4101\n",
      "Epoch 25: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 607us/sample - loss: 1.4101 - val_loss: 1.3663\n",
      "Epoch 26/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4101\n",
      "Epoch 26: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 670us/sample - loss: 1.4101 - val_loss: 1.3726\n",
      "Epoch 27/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4077\n",
      "Epoch 27: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 680us/sample - loss: 1.4077 - val_loss: 1.3741\n",
      "Epoch 28/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.4043\n",
      "Epoch 28: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 694us/sample - loss: 1.4043 - val_loss: 1.3685\n",
      "Epoch 29/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3963\n",
      "Epoch 29: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 632us/sample - loss: 1.3963 - val_loss: 1.3688\n",
      "Epoch 30/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3943\n",
      "Epoch 30: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 605us/sample - loss: 1.3943 - val_loss: 1.3659\n",
      "Epoch 31/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3908\n",
      "Epoch 31: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 674us/sample - loss: 1.3908 - val_loss: 1.3707\n",
      "Epoch 32/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3854\n",
      "Epoch 32: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 643us/sample - loss: 1.3854 - val_loss: 1.3658\n",
      "Epoch 33/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3843\n",
      "Epoch 33: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 639us/sample - loss: 1.3843 - val_loss: 1.3619\n",
      "Epoch 34/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3826\n",
      "Epoch 34: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 703us/sample - loss: 1.3826 - val_loss: 1.3612\n",
      "Epoch 35/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3788\n",
      "Epoch 35: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 697us/sample - loss: 1.3788 - val_loss: 1.3583\n",
      "Epoch 36/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3780\n",
      "Epoch 36: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 673us/sample - loss: 1.3780 - val_loss: 1.3627\n",
      "Epoch 37/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3763\n",
      "Epoch 37: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 603us/sample - loss: 1.3763 - val_loss: 1.3592\n",
      "Epoch 38/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3755\n",
      "Epoch 38: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 609us/sample - loss: 1.3755 - val_loss: 1.3588\n",
      "Epoch 39/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3722\n",
      "Epoch 39: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 600us/sample - loss: 1.3722 - val_loss: 1.3594\n",
      "Epoch 40/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3730\n",
      "Epoch 40: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 657us/sample - loss: 1.3730 - val_loss: 1.3550\n",
      "Epoch 41/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3690\n",
      "Epoch 41: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 701us/sample - loss: 1.3690 - val_loss: 1.3538\n",
      "Epoch 42/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3682\n",
      "Epoch 42: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 699us/sample - loss: 1.3682 - val_loss: 1.3570\n",
      "Epoch 43/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3700\n",
      "Epoch 43: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 21s 666us/sample - loss: 1.3700 - val_loss: 1.3574\n",
      "Epoch 44/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3655\n",
      "Epoch 44: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 644us/sample - loss: 1.3655 - val_loss: 1.3543\n",
      "Epoch 45/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3640\n",
      "Epoch 45: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 624us/sample - loss: 1.3640 - val_loss: 1.3577\n",
      "Epoch 46/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3643\n",
      "Epoch 46: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 632us/sample - loss: 1.3643 - val_loss: 1.3502\n",
      "Epoch 47/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3592\n",
      "Epoch 47: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 19s 610us/sample - loss: 1.3592 - val_loss: 1.3566\n",
      "Epoch 48/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3611\n",
      "Epoch 48: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 693us/sample - loss: 1.3611 - val_loss: 1.3546\n",
      "Epoch 49/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3618\n",
      "Epoch 49: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 22s 693us/sample - loss: 1.3618 - val_loss: 1.3542\n",
      "Epoch 50/50\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3582\n",
      "Epoch 50: val_loss did not improve from 1.31827\n",
      "31555/31555 [==============================] - 20s 622us/sample - loss: 1.3582 - val_loss: 1.3488\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:18:00.224692: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_9_5/lstm_cell_194/bias/Assign' id:97625 op device:{requested: '', assigned: ''} def:{{{node lstm_9_5/lstm_cell_194/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_9_5/lstm_cell_194/bias, lstm_9_5/lstm_cell_194/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:18:05.262040: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_18_5/lstm_cell_203/bias/v/Assign' id:103691 op device:{requested: '', assigned: ''} def:{{{node lstm_18_5/lstm_cell_203/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_18_5/lstm_cell_203/bias/v, lstm_18_5/lstm_cell_203/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:18:10.173506: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1932)\n",
      "(1514, 1932)\n",
      "(1644, 1932)\n",
      "(1764, 1932)\n",
      "(1836, 1932)\n",
      "(1699, 1932)\n",
      "(1369, 1932)\n",
      "(1766, 1932)\n",
      "(1631, 1932)\n",
      "(1692, 1932)\n",
      "(1550, 1932)\n",
      "(1920, 1932)\n",
      "(1739, 1932)\n",
      "(1788, 1932)\n",
      "(1812, 1932)\n",
      "(1824, 1932)\n",
      "(970, 1932)\n",
      "(1656, 1932)\n",
      "(1896, 1932)\n",
      "{1: 5.938984860556306, 2: 4.444939730169126, 4: 10.0, 5: 6.03415506314621, 6: 6.156892604880481, 8: 8.929129590361804, 9: 6.466825506709358, 10: 8.068773406916957, 11: 6.958715698797419, 12: 8.97306475025023, 13: 6.79334471169043, 17: 9.491106735796324, 19: 9.20331950461409, 21: 8.990616782366125, 22: 1.0, 26: 5.361113578926464, 27: 4.467206575932888, 28: 5.886684945613486, 29: 1.8327548785479548}\n",
      "Train on 31555 samples, validate on 3514 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:22:54.294737: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 9.2737\n",
      "Epoch 1: val_loss improved from inf to 1.33844, saving model to ./checkpoints/unknown_person_p25_22.h5\n",
      "31555/31555 [==============================] - 33s 1ms/sample - loss: 9.2737 - val_loss: 1.3384\n",
      "Epoch 2/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.2185\n",
      "Epoch 2: val_loss improved from 1.33844 to 1.33676, saving model to ./checkpoints/unknown_person_p25_22.h5\n",
      "31555/31555 [==============================] - 19s 600us/sample - loss: 9.2185 - val_loss: 1.3368\n",
      "Epoch 3/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.3241\n",
      "Epoch 3: val_loss did not improve from 1.33676\n",
      "31555/31555 [==============================] - 19s 607us/sample - loss: 9.3241 - val_loss: 1.3429\n",
      "Epoch 4/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.2067\n",
      "Epoch 4: val_loss did not improve from 1.33676\n",
      "31555/31555 [==============================] - 19s 604us/sample - loss: 9.2067 - val_loss: 1.3437\n",
      "Epoch 5/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.1295\n",
      "Epoch 5: val_loss improved from 1.33676 to 1.33178, saving model to ./checkpoints/unknown_person_p25_22.h5\n",
      "31555/31555 [==============================] - 21s 676us/sample - loss: 9.1295 - val_loss: 1.3318\n",
      "Epoch 6/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0920\n",
      "Epoch 6: val_loss improved from 1.33178 to 1.32351, saving model to ./checkpoints/unknown_person_p25_22.h5\n",
      "31555/31555 [==============================] - 22s 708us/sample - loss: 9.0920 - val_loss: 1.3235\n",
      "Epoch 7/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.1076\n",
      "Epoch 7: val_loss did not improve from 1.32351\n",
      "31555/31555 [==============================] - 21s 653us/sample - loss: 9.1076 - val_loss: 1.3262\n",
      "Epoch 8/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0856\n",
      "Epoch 8: val_loss improved from 1.32351 to 1.32083, saving model to ./checkpoints/unknown_person_p25_22.h5\n",
      "31555/31555 [==============================] - 22s 702us/sample - loss: 9.0856 - val_loss: 1.3208\n",
      "Epoch 9/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0773\n",
      "Epoch 9: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 22s 705us/sample - loss: 9.0773 - val_loss: 1.3263\n",
      "Epoch 10/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0716\n",
      "Epoch 10: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 22s 685us/sample - loss: 9.0716 - val_loss: 1.3304\n",
      "Epoch 11/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0418\n",
      "Epoch 11: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 21s 679us/sample - loss: 9.0418 - val_loss: 1.3224\n",
      "Epoch 12/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0370\n",
      "Epoch 12: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 22s 683us/sample - loss: 9.0370 - val_loss: 1.3246\n",
      "Epoch 13/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0241\n",
      "Epoch 13: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 22s 699us/sample - loss: 9.0241 - val_loss: 1.3296\n",
      "Epoch 14/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0289\n",
      "Epoch 14: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 22s 700us/sample - loss: 9.0289 - val_loss: 1.3242\n",
      "Epoch 15/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0107\n",
      "Epoch 15: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 20s 640us/sample - loss: 9.0107 - val_loss: 1.3230\n",
      "Epoch 16/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0205\n",
      "Epoch 16: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 19s 604us/sample - loss: 9.0205 - val_loss: 1.3273\n",
      "Epoch 17/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0267\n",
      "Epoch 17: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 19s 600us/sample - loss: 9.0267 - val_loss: 1.3285\n",
      "Epoch 18/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9940\n",
      "Epoch 18: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 19s 598us/sample - loss: 8.9940 - val_loss: 1.3245\n",
      "Epoch 19/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 9.0074\n",
      "Epoch 19: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 20s 619us/sample - loss: 9.0074 - val_loss: 1.3237\n",
      "Epoch 20/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 8.9981\n",
      "Epoch 20: val_loss did not improve from 1.32083\n",
      "31555/31555 [==============================] - 21s 681us/sample - loss: 8.9981 - val_loss: 1.3266\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:30:03.139268: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_20_6/lstm_cell_242/recurrent_kernel/Assign' id:118774 op device:{requested: '', assigned: ''} def:{{{node lstm_20_6/lstm_cell_242/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_20_6/lstm_cell_242/recurrent_kernel, lstm_20_6/lstm_cell_242/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:30:09.512833: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_8_6/lstm_cell_230/kernel/v/Assign' id:122929 op device:{requested: '', assigned: ''} def:{{{node lstm_8_6/lstm_cell_230/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_8_6/lstm_cell_230/kernel/v, lstm_8_6/lstm_cell_230/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31555 samples, validate on 3514 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:30:18.872799: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:30:51.273955: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3211"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:31:14.383457: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.32110, saving model to ./checkpoints/unknown_person_baseline_2_p25_22.h5\n",
      "31555/31555 [==============================] - 36s 1ms/sample - loss: 1.3211 - val_loss: 1.3211\n",
      "Epoch 2/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3182\n",
      "Epoch 2: val_loss improved from 1.32110 to 1.32043, saving model to ./checkpoints/unknown_person_baseline_2_p25_22.h5\n",
      "31555/31555 [==============================] - 21s 654us/sample - loss: 1.3182 - val_loss: 1.3204\n",
      "Epoch 3/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3466\n",
      "Epoch 3: val_loss did not improve from 1.32043\n",
      "31555/31555 [==============================] - 20s 648us/sample - loss: 1.3466 - val_loss: 1.3215\n",
      "Epoch 4/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3295\n",
      "Epoch 4: val_loss improved from 1.32043 to 1.31520, saving model to ./checkpoints/unknown_person_baseline_2_p25_22.h5\n",
      "31555/31555 [==============================] - 19s 617us/sample - loss: 1.3295 - val_loss: 1.3152\n",
      "Epoch 5/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3237\n",
      "Epoch 5: val_loss did not improve from 1.31520\n",
      "31555/31555 [==============================] - 19s 609us/sample - loss: 1.3237 - val_loss: 1.3171\n",
      "Epoch 6/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3210\n",
      "Epoch 6: val_loss did not improve from 1.31520\n",
      "31555/31555 [==============================] - 19s 605us/sample - loss: 1.3210 - val_loss: 1.3163\n",
      "Epoch 7/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3197\n",
      "Epoch 7: val_loss improved from 1.31520 to 1.31437, saving model to ./checkpoints/unknown_person_baseline_2_p25_22.h5\n",
      "31555/31555 [==============================] - 21s 652us/sample - loss: 1.3197 - val_loss: 1.3144\n",
      "Epoch 8/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3204\n",
      "Epoch 8: val_loss did not improve from 1.31437\n",
      "31555/31555 [==============================] - 20s 628us/sample - loss: 1.3204 - val_loss: 1.3148\n",
      "Epoch 9/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3169\n",
      "Epoch 9: val_loss did not improve from 1.31437\n",
      "31555/31555 [==============================] - 19s 602us/sample - loss: 1.3169 - val_loss: 1.3172\n",
      "Epoch 10/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3173\n",
      "Epoch 10: val_loss did not improve from 1.31437\n",
      "31555/31555 [==============================] - 19s 604us/sample - loss: 1.3173 - val_loss: 1.3178\n",
      "Epoch 11/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3131\n",
      "Epoch 11: val_loss improved from 1.31437 to 1.31353, saving model to ./checkpoints/unknown_person_baseline_2_p25_22.h5\n",
      "31555/31555 [==============================] - 20s 647us/sample - loss: 1.3131 - val_loss: 1.3135\n",
      "Epoch 12/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3164\n",
      "Epoch 12: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 21s 677us/sample - loss: 1.3164 - val_loss: 1.3220\n",
      "Epoch 13/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3134\n",
      "Epoch 13: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 22s 687us/sample - loss: 1.3134 - val_loss: 1.3207\n",
      "Epoch 14/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3137\n",
      "Epoch 14: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 22s 692us/sample - loss: 1.3137 - val_loss: 1.3285\n",
      "Epoch 15/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3163\n",
      "Epoch 15: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 21s 656us/sample - loss: 1.3163 - val_loss: 1.3153\n",
      "Epoch 16/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3126\n",
      "Epoch 16: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 20s 623us/sample - loss: 1.3126 - val_loss: 1.3175\n",
      "Epoch 17/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3112\n",
      "Epoch 17: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 20s 632us/sample - loss: 1.3112 - val_loss: 1.3167\n",
      "Epoch 18/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3112\n",
      "Epoch 18: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 20s 620us/sample - loss: 1.3112 - val_loss: 1.3165\n",
      "Epoch 19/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3108\n",
      "Epoch 19: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 19s 613us/sample - loss: 1.3108 - val_loss: 1.3199\n",
      "Epoch 20/20\n",
      "31555/31555 [==============================] - ETA: 0s - loss: 1.3089\n",
      "Epoch 20: val_loss did not improve from 1.31353\n",
      "31555/31555 [==============================] - 21s 676us/sample - loss: 1.3089 - val_loss: 1.3148\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [25]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
